rustls/tests/api.rs

```diff
@@ -3293,7 +3293,7 @@ mod test_quic {
             payload: MessagePayload::Handshake(HandshakeMessagePayload {
                 typ: HandshakeType::ClientHello,
                 payload: HandshakePayload::ClientHello(ClientHelloPayload {
-                    client_version: ProtocolVersion::TLSv1_3,
+                    client_version: ProtocolVersion::TLSv1_2,
```

Comment:
This issue would may have been avoided if `client_version` was instead named `legacy_version` as in the TLS 1.3 spec. We should (ideally in a separate PR) rename these fields to their TLS 1.3 names.

Reply:
fwiw, I got confused by this name while working on ECH, so I agree here.